lib/core/test/dict_test.cpp

```diff
@@ -0,0 +1,557 @@
+}
+
+TEST(Dict, key_iterator_throws_if_element_erased_front) {
+  DimDict dict{{Dim::Y, -4122}, {Dim::Row, 5619}, {Dim::Event, 829}};
+  auto it = dict.keys_begin();
+  dict.erase(Dim::Y);
+  EXPECT_THROW_DISCARD(*it, std::runtime_error);
+  EXPECT_THROW_DISCARD(++it, std::runtime_error);
+  EXPECT_THROW_DISCARD(it == dict.keys_end(), std::runtime_error);
+}
+
+TEST(Dict, key_iterator_throws_if_element_erased_middle) {
+  DimDict dict{{Dim::Y, -4122}, {Dim::Row, 5619}, {Dim::Event, 829}};
+  auto it = dict.keys_begin();
+  dict.erase(Dim::Row);
+  EXPECT_THROW_DISCARD(*it, std::runtime_error);
+  EXPECT_THROW_DISCARD(++it, std::runtime_error);
+  EXPECT_THROW_DISCARD(it == dict.keys_end(), std::runtime_error);
+}
+
+TEST(Dict, key_iterator_throws_if_element_erased_back) {
+  DimDict dict{{Dim::Y, -4122}, {Dim::Row, 5619}, {Dim::Event, 829}};
+  auto it = dict.keys_begin();
+  dict.erase(Dim::Event);
+  EXPECT_THROW_DISCARD(*it, std::runtime_error);
+  EXPECT_THROW_DISCARD(++it, std::runtime_error);
+  EXPECT_THROW_DISCARD(it == dict.keys_end(), std::runtime_error);
+}
+
+TEST(Dict, key_iterator_throws_if_element_erased_front_after_increment) {
+  DimDict dict{{Dim::Y, -4122}, {Dim::Row, 5619}, {Dim::Event, 829}};
+  auto it = dict.keys_begin();
+  ++it;
+  dict.erase(Dim::Y);
+  EXPECT_THROW_DISCARD(*it, std::runtime_error);
+  EXPECT_THROW_DISCARD(++it, std::runtime_error);
+  EXPECT_THROW_DISCARD(it == dict.keys_end(), std::runtime_error);
+}
+
+TEST(Dict, key_iterator_throws_if_element_erased_middle_after_increment) {
+  DimDict dict{{Dim::Y, -4122}, {Dim::Row, 5619}, {Dim::Event, 829}};
+  auto it = dict.keys_begin();
+  ++it;
+  dict.erase(Dim::Row);
+  EXPECT_THROW_DISCARD(*it, std::runtime_error);
+  EXPECT_THROW_DISCARD(++it, std::runtime_error);
+  EXPECT_THROW_DISCARD(it == dict.keys_end(), std::runtime_error);
+}
+
+TEST(Dict, key_iterator_throws_if_element_erased_back_after_increment) {
+  DimDict dict{{Dim::Y, -4122}, {Dim::Row, 5619}, {Dim::Event, 829}};
+  auto it = dict.keys_begin();
+  ++it;
+  dict.erase(Dim::Event);
+  EXPECT_THROW_DISCARD(*it, std::runtime_error);
+  EXPECT_THROW_DISCARD(++it, std::runtime_error);
+  EXPECT_THROW_DISCARD(it == dict.keys_end(), std::runtime_error);
+}
+
+TEST(Dict, key_iterator_does_not_throw_if_created_after_modification) {
+  DimDict dict{{Dim::Event, 94196}};
+  dict.insert_or_assign(Dim::X, -8164);
+  EXPECT_EQ(dict.begin()->second, 94196);
+  dict.erase(Dim::Event);
+  EXPECT_EQ(dict.begin()->second, -8164);
+}
+
+TEST(Dict, value_iterator_produces_correct_values) {
+  DimDict dict{{Dim::Time, 61892}, {Dim::Event, 619}};
+  auto it = dict.values_begin();
+  EXPECT_EQ(*it, 61892);
+  ++it;
+  EXPECT_EQ(*it, 619);
+  ++it;
+  EXPECT_EQ(it, dict.values_end());
+}
+
+TEST(Dict, const_value_iterator_produces_correct_values) {
+  DimDict dict{{Dim::Time, 4561}, {Dim::Event, 76}};
+  // NOLINTNEXTLINE(performance-unnecessary-copy-initialization)
```

Comment:
Can we globally disable all "performance" linting within the `test/` subfolders?

Reply:
Makes sense